plugins/module_utils/job.py

```diff
@@ -122,430 +114,190 @@ def _job_not_found(job_id, owner, job_name, dd_name, ovrr=None):
     """
+    stp = []
+    if "STEP WAS EXECUTED" in job_str:
+        pile = re.findall(r"(.*?)\s-\sSTEP\sWAS\sEXECUTED\s-\s(.*?)\n", job_str)
+        for match in pile:
+            st = {
+                "step_name": match[0].split()[-1],
+                "step_cc": match[1].split()[-1],
+            }
+            stp.append(st)
 
-    rc = None
-    match = re.search(r"\s*CC\s*([0-9]+)", rc_str)
-    if match:
-        rc = int(match.group(1))
-    return rc
-
-
-def _get_return_code_str(rc_str):
-    """Parse an integer return code from
-    z/OS job output return code string.
+    return stp
 
-    Arguments:
-        rc_str {str} -- The return code message from z/OS job log (eg. "CC 0000" or "ABEND")
 
-    Returns:
-        Union[str, NoneType] -- Returns string RC or ABEND code if possible, if not returns NoneType
-    """
-    rc = None
-    match = re.search(
-        r"(?:\s*CC\s*([0-9]+))|(?:ABEND\s*((?:S|U)[0-9]+)|(?:JCL ERROR))", rc_str
-    )
-    if match:
-        rc = match.group(1) or match.group(2)
-    return rc
+def _zget_job_status(job_id="*", owner="*", job_name="*", dd_name=None):
+    if job_id == "*":
+        job_query = None
+    else:
+        job_query = job_id
+
+    # entries = listing(job_query, owner)   1.2.0 has owner paramn, 1.1 does not
+    entries = listing(job_query)
+
+    final_entries = []
+    if entries:
+        for entry in entries:
+            if owner != "*":
+                if owner != entry.owner:
+                    continue
+            if job_name != "*":
+                if job_name != entry.name:
+                    continue
+
+            job = {}
+
+            job["job_id"] = entry.id
+            job["job_name"] = entry.name
+            job["subsystem"] = ""
+            job["system"] = ""
+            job["owner"] = entry.owner
+
+            job["ret_code"] = {}
+            job["ret_code"]["msg"] = entry.status + " " + entry.rc
+            job["ret_code"]["msg_code"] = entry.rc
+
+            job["ret_code"]["code"] = ""
+            if len(entry.rc) > 0:
+                if entry.rc.isdigit():
+                    job["ret_code"]["code"] = int(entry.rc)
+
+            job["ret_code"]["msg_text"] = entry.status
+
+            job["class"] = ""
+            job["content_type"] = ""
+
+            job["ret_code"]["steps"] = []
+            job["ddnames"] = []
+            list_of_dds = list_dds(entry.id)
+
+            for single_dd in list_of_dds:
+                dd = {}
+                if dd_name is not None:
+                    if dd_name not in single_dd["dataset"]:
+                        continue
+
+                if "dataset" not in single_dd:
+                    continue
+
+                dd["ddname"] = single_dd["dataset"]
+                if "recnum" in single_dd:
+                    dd["record_count"] = single_dd["recnum"]
+                else:
+                    dd["record_count"] = "n/a"
```

Comment:
If there is no record count (not sure what case this would be, maybe job has not executed) then is "n/a" a good response? I am thinking None for the pythonic way or "" for the non breaking way but I think our users understand the None paradigm.

Reply:
We don't use the record count internally, so I've switched it to None.